homeassistant/components/binary_sensor/xboxlive.py

```diff
@@ -0,0 +1,172 @@
+})
+
+
+# pylint: disable=unused-argument
+@asyncio.coroutine
+def async_setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Xbox platform."""
+    import xbox
+    xbox.client.authenticate(login=config.get(CONF_USERNAME),
+                             password=config.get(CONF_PASSWORD))
+    devices = []
+
+    for xuid in config.get(CONF_XUIDS):
+        try:
+            gamer_profile = xbox.GamerProfile.from_xuid(xuid)
+        except xbox.exceptions.GamertagNotFound:
+            _LOGGER.error("Can not found Xbox UID %s", xuid)
+            continue
+        devices.append(XboxSensor(hass, gamer_profile))
+
+    for gamertag in config.get(CONF_GAMERTAGS):
+        try:
+            gamer_profile = xbox.GamerProfile.from_gamertag(gamertag)
+        except xbox.exceptions.GamertagNotFound:
+            _LOGGER.error("Can not found Gamertag %s", gamertag)
+            continue
+        devices.append(XboxSensor(hass, gamer_profile))
+
+    if devices:
+        add_devices(devices)
+    else:
+        return False
+
+
+class XboxSensor(BinarySensorDevice):
+    """A class for the Xbox account."""
+
+    def __init__(self, hass, gamer_profile):
+        """Initialize the sensor."""
+        self._hass = hass
+        self._state = STATE_UNKNOWN
+        self._gamer_profile = gamer_profile
+        self._gamertag = gamer_profile.gamertag
+        self._gamerscore = None
+        self._devices = []
+        self._xuid = gamer_profile.xuid
+        self._picture = gamer_profile.gamerpic
+        self._headers = {
+            'x-xbl-contract-version': '2',
+            'User-Agent': ('XboxRecord.Us Like SmartGlass/2.105.0415 '
+                           'CFNetwork/711.3.18 Darwin/14.0.0')}
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "xboxlive_{}".format(self._gamertag)
+
+    @property
+    def is_on(self):
+        """Return the status of the sensor."""
+        return self._state == 'online'
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attributes = {
+            "xuid": self._xuid,
+            "gamerscore": self._gamerscore,
+        }
+        for device in self._devices:
+            for title in device.get('titles'):
+                attributes[
+                    '{}_{}'.format(device.get('type').lower(),
+                                   title.get('placement').lower())
+                ] = title.get('name')
+
+        return attributes
+
+    @property
+    def entity_picture(self):
+        """Avatar of the account."""
+        return self._picture
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return ICON
+
+    @asyncio.coroutine
+    def async_update(self):
```

Comment:
The xbox library doesn't seem to use asyncio. So then you should not use it here either, since all coroutines are run within the event loop. Make this a regular `update` method that is not a coroutine.